src/mongooseim.app.src

```diff
@@ -1,57 +1,47 @@
-%% $Id$
-
 {application, mongooseim,
  [{description, "MongooseIM"},
   {vsn, {cmd, "tools/generate_vsn.sh"}},
   {modules, []},
   {registered, []},
-  {applications, [
-                  asn1,
-                  backoff,
-                  cache_tab,
-                  cowboy,
-                  cpool,
-                  crypto,
-                  fast_scram,
-                  eredis,
-                  eldap,
-                  exml,
-                  exometer_core,
-                  exometer_report_graphite,
-                  exometer_report_statsd,
-                  graphql,
-                  jiffy,
-                  jwerl,
-                  fusco,
-                  idna,
-                  kernel,
-                  lasse,
-                  observer_cli,
-                  pa,
-                  public_key,
-                  prometheus_cowboy,
-                  ranch,
-                  recon,
-                  runtime_tools,
-                  sasl,
-                  ssl,
-                  stdlib,
-%%                  syslogger, % uncomment to enable a logger handler for syslog
-                  jid,
-                  tirerl,
-                  uuid,
-                  xmerl,
-                  worker_pool,
-                  gen_fsm_compat,
-                  erl_csv,
-                  trails,
-                  cowboy_swagger,
-                  tomerl,
-                  flatlog,
-                  segmented_cache,
-                  opuntia
-                 ]},
-  {included_applications, [mnesia, cets, erlang_doctor]},
+  {applications,
```

Comment:
What logic was applied when reworking this whole list? Previously it was partially sorted alphabetically, partially not, but now it is divided into three parts for reasons I can't figure out. It's also shorter - why?

Reply:
Here we only need the apps that must be **started** _before_ MongooseIM does. The order is the order in which they'll be started, so `kernel` and `stdlib` must go first (in reality _any_ of the included apps already required kernel and stdlib much before MIM even came so it's "fine"). The rest of the order is really not any special choice, first I put what comes with OTP and later all 3rd-party libraries, more or less. They should obey an order of priorities because that's the order in which OTP will start dependencies, but for the most part it's just informal.

It's shorter because we're removing from here all the apps that we don't want to be started before MIM, that's why we're moving the starting logic to explicit `ensure_all_started/2` down the road.

Reply:
Also, `included_applications` does not play well with rebar3, I tried that, if you declare here an included app that happen to be a dependency of any of the top apps, and the _exact_ version doesn't match, `rebar3` fails compilation.